promdump/promdump.go

```diff
@@ -651,9 +770,52 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		var filename string
+		filename = *tarFilename
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			filename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(filename); err == nil {
+				log.Fatalf("Error: %s already exists. Please choose a different filename.\n", filename)
+				return
+			}
+		}
+
+		out, err := os.Create(filename)
+		if err != nil {
+			log.Printf("Error writing archive: %v (File: %s)\n", err, filename)
+			return
+		}
+
+		// Create the archive
+		err = createArchive(out)
+		if err != nil {
+			log.Printf("Error creating archive: %v\n", err)
+		}
+		// cleaning files
+		for _, v := range collectMetrics {
+			metricName, err := getMetricName(v)
+			if err != nil {
+				log.Fatalf("Error removing files: %v", err)
```

Comment:
Be more specific about where this error is coming from. Something like `log.Fatalf("could not retrieve metric name for metric `v`: %v", err)`.

Reply:
change the sentence as suggested ,  but use a warning instead of a fatal as suggested from above comment